megatron/core/transformer/attention.py

```diff
@@ -658,10 +662,14 @@ def flash_decode_and_prefill(
                     "cache_seqlens": seqlens_k,
                     "causal": True,
                     "page_table" if HAVE_FA3 else "block_table": block_table,
+                    "num_splits": 0 if not self.batch_invariant_mode else 1,
                 }
                 if HAVE_FA3:
                     output_total = flash_attn3_with_kvcache(**flash_attn_args)
                 else:
+                    assert (
+                        not self.batch_invariant_mode
+                    ), "Batch invariant mode is not supported for flash attention 2"
```

Comment:
With the FA3 deterministic-mode issue caused by a randomly-initialized (`torch.empty`) KV$ that's been discussed in inference calls, is there a possible danger with relying on FA3 here?